### PR TITLE
Fhircore enhancements

### PR DESCRIPTION
**Pull request by ndegwamartin**
**In this PR**
- Align Sentry DSN configuration properties with existing configs
- Add Sentry scope metadata for troubleshooting
- Disable clear text traffic 
- Refactor PIN String variable to CharArray
- Fix CI Build - Fixes broken unit test `FhirCarePlanGeneratorTest`

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — DISMISSED** [review]

**Review by hilpitome — DISMISSED** [review]
lgtm

**Review by pld — DISMISSED** [review]

**Comment by pld** [comment]
not sure what's up the tests seem to pass locally

File: android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt [hunk 1]
```diff
@@ -83,7 +86,13 @@ class QuestApplication : Application(), DataCaptureConfig.Provider, Configuratio
             enableAutoFragmentLifecycleTracing = true
           )
         )
+        try {
+          options.environment = URL(BuildConfig.FHIR_BASE_URL)?.getSubDomain()?.replace('-', '.')
+        } catch (e: Exception) {
+          Timber.e(e)
+        }
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt:93]: Is this try-catch necessary?

Comment (ndegwamartin) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt:93]: Was part of trying to troubleshoot which code updates introduced failing the Unit Test system

Comment (ndegwamartin) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt:93]: Reverted

**Review by pld — APPROVED** [review]

**Pull request merged**